python/whylogs/core/constraints/factories/distribution_metrics.py

```diff
@@ -73,19 +112,14 @@ def mean_between_range(column_name: str, lower: float, upper: float, skip_missin
         Upper bound of the value range
     skip_missing: bool
         If skip_missing is True, missing distribution metrics will make the check pass.
-        If False, the check will fail on missing metrics
+        If False, the check will fail on missing metrics, such as on an empty dataset
     """
 
-    def is_mean_between(metric) -> bool:
-        if not metric.kll.value.is_empty():
-            return lower <= metric.avg <= upper
-        else:
-            return True if skip_missing else False
-
     constraint = MetricConstraint(
         name=f"{column_name} mean between {lower} and {upper} (inclusive)",
-        condition=is_mean_between,
+        condition=lambda x: lower <= x.avg <= upper,
```

Comment:
same, why lambda

Reply:
replaced lambda to Predicate in latest commit